stdlib/LinearAlgebra/src/lapack.jl

```diff
@@ -7045,4 +7045,48 @@ julia> LAPACK.lacpy!(B, A, 'U')
 """
 lacpy!(B::AbstractMatrix, A::AbstractMatrix, uplo::AbstractChar)
 
+# Routines for column permutation
+for (fn, elty) in ((:slapmt_, :Float32),
+                (:dlapmt_, :Float64),
+                (:clapmt_, :ComplexF32),
+                (:zlapmt_, :ComplexF64))
+    @eval begin
+        function lapmt!(A::AbstractMatrix{$elty}, p::AbstractVector{<:BlasInt}, fb::Bool)
+            require_one_based_indexing(A)
+            chkstride1(A)
+            m, n = size(A)
+            lda = max(1, stride(A, 2))
+            lp = length(p)
+            if n != lp
+                throw(DimensionMismatch("The second dimension of A, ($m,$n), and P, ($lp), must match"))
+            end
+            ccall((@blasfunc($fn), libblastrampoline), Cvoid,
+            (Ref{BlasInt}, Ref{BlasInt}, Ref{BlasInt}, Ptr{Float64}, Ref{BlasInt}, Ptr{BlasInt}),
+            fb, m, n, A, lda, p) ;
+        end
+    end
+end
+
+# Routines for row permutation
+for (fn, elty) in ((:slapmr_, :Float32),
+                (:dlapmr_, :Float64),
+                (:clapmr_, :ComplexF32),
+                (:zlapmr_, :ComplexF64))
+    @eval begin
+        function lapmr!(A::AbstractMatrix{$elty}, p::AbstractVector{<:BlasInt}, fb::Bool)
+            require_one_based_indexing(A)
+            chkstride1(A)
+            m, n = size(A)
+            lda = max(1, stride(A, 2))
+            lp = length(p)
+            if m != lp
+                throw(DimensionMismatch("The first dimension of A, ($m,$n), and P, ($lp), must match"))
+            end
+            ccall((@blasfunc($fn), libblastrampoline), Cvoid,
+            (Ref{BlasInt}, Ref{BlasInt}, Ref{BlasInt}, Ptr{Float64}, Ref{BlasInt}, Ptr{BlasInt}),
+            fb, m, n, A, lda, p) ;
+        end
+    end
+end
+
```

Comment:
Do we still need these? Given the generic ones we have, having these adds testing and maintenance burden.

Reply:
Thank you very much, @dkarrasch. I'm truly indebted to your support and guidance. Indeed, these can go, and I will remove them soon. Soon after, I'll rerun the benchmark tests, analyse a few things, update NEWS.md, and let you know as soon as things are finalised!